contribs/emissions/src/main/java/org/matsim/contrib/emissions/PositionEmissionsModule.java

```diff
@@ -189,31 +195,33 @@ private void handlePositionEvent(PositionEvent event) {
 			trajectories.computeIfAbsent(event.getVehicleId(), key -> new LinkedList<>()).add(event);
 		}
 
-		private void computeFirstColdEmissionEvent(PositionEvent event) {
+		private Map<Pollutant, Double> computeColdEmissions(PositionEvent event, Vehicle vehicle, double distanceToLastPosition) {
 
-			var vehicle = getVehicle(event.getVehicleId());
-			var startEngineTime = vehiclesInTraffic.get(vehicle.getId()).getTime();
-			var emissions = emissionCalculator.calculateColdEmissions(vehicle, event.getLinkId(),
-					startEngineTime, parkingDurations.get(vehicle.getId()), 1);
+			// we remember the vehicles which are currently emmitting cold emissions if not stored here return nothing
+			if (!vehiclesEmittingColdEmissions.contains(vehicle.getId()))
+				return emissionCalculator.getEmptyColdEmissions();
 
-			var emisPosEv = new PositionEmissionEvent(event, emissions, "cold");
-			eventsManager.processEvent(emisPosEv);
-		}
-
-		private void computeSecondColdEmissionEvent(PositionEvent event) {
-			// check for cold emissions which should be computed once if distance is > 1000m
 			double distance = calculateTravelledDistance(event);
 
-			if (distance > 1000) {
-				var vehicle = getVehicle(event.getVehicleId());
-				var emissions = emissionCalculator.calculateColdEmissions(vehicle, event.getLinkId(),
-						event.getTime(), parkingDurations.get(vehicle.getId()), 2);
-				var emisPosEv = new PositionEmissionEvent(event, emissions, "cold");
-				eventsManager.processEvent(emisPosEv);
-
-				// this makes sure that this is only computed once
-				vehiclesWaitingForSecondColdEmissionEvent.remove(vehicle.getId());
+			// this model assumes vehicles to emmit cold emissions for the first 2000m of a trip remove a vehicle from
+			// the list of emmiting vehicles if the current trajectory is longer than 2000m
+			if (distance > 2000) {
+				vehiclesEmittingColdEmissions.remove(vehicle.getId());
+				return emissionCalculator.getEmptyColdEmissions();
 			}
+
+			// HBEFA assumes a constantly decreasing ammount of cold emissions depending on the distance travelled
+			// the underlying emission module simplifies this into two steps. Between 0-1km and 1-2km. We use the same
+			// classes here because we don't want to rewrite all the stuff. The cold emission module computes emissions
+			// for 1000m. We take these as is and muliply with distanceToLastPosition / 1000. This way we have the fraction
+			// of cold emissions for the distance travelled during the last time step janek oct' 2021
```

Comment:
Comment for the comment: HBEFA provides the data in these two bins: 01-km and 1-2 km. So it is not our decision to use these classes ;)

Reply:
Nice. I read their documentation but obviously missed the part where they described the abstraction of two distance classes. I guess this is even better since one doesn't have to discuss this simplification since it happened upstream.